test/performance/odp_packet_gen.c

```diff
@@ -1451,63 +1479,96 @@ static int init_packets(test_global_t *global, int pktio,
 		ip->tot_len = odp_cpu_to_be_16(pkt_len - l2_len);
 		ip->id = odp_cpu_to_be_16(seq + i);
 		ip->ttl = 64;
-		ip->proto = ODPH_IPPROTO_UDP;
+		ip->proto = use_tcp ? ODPH_IPPROTO_TCP : ODPH_IPPROTO_UDP;
 		ip->src_addr = odp_cpu_to_be_32(test_options->ipv4_src);
 		ip->dst_addr = odp_cpu_to_be_32(test_options->ipv4_dst);
 		ip->chksum = ~odp_chksum_ones_comp16(ip, ODPH_IPV4HDR_LEN);
 
-		/* UDP */
-		udp = (odph_udphdr_t *)((uint8_t *)data + l2_len +
-		      ODPH_IPV4HDR_LEN);
-		memset(udp, 0, ODPH_UDPHDR_LEN);
-		udp->src_port = odp_cpu_to_be_16(udp_src);
-		udp->dst_port = odp_cpu_to_be_16(udp_dst);
-		udp->length   = odp_cpu_to_be_16(payload_len + ODPH_UDPHDR_LEN);
-		udp->chksum   = 0;
+		u8 = ((uint8_t *)data + l2_len + ODPH_IPV4HDR_LEN);
+
+		if (use_tcp) {
+			odph_tcphdr_t *tcp = (odph_tcphdr_t *)u8;
+
+			memset(tcp, 0, ODPH_TCPHDR_LEN);
+			tcp->src_port = odp_cpu_to_be_16(src_port);
+			tcp->dst_port = odp_cpu_to_be_16(dst_port);
+			tcp->seq_no   = odp_cpu_to_be_32(tcp_seqnum);
+			tcp->ack_no   = odp_cpu_to_be_32(0x12345678);
+			tcp->window   = odp_cpu_to_be_16(0x4000);
+			tcp->hl       = 5;
+			tcp->ack      = 1;
```

Comment:
If it doesn't matter what kind of TCP packets program is generating, I guess they could be `SYN` packets, would avoid faking any session state maintenance.

(Program would be basically `SYN` flooding, but does it matter?)

Reply:
The target was to create "valid enough" packets for Wireshark, so in the next phase it can be used to validate LSO transmitted packets. Setting the SYN flag seems to make Wireshark unhappy: "TCP Port numbers reused".